chore: refactor aws eks install page - reduce nodes, make ubuntu default (#1077)
coder/docs: setup/kubernetes/aws.md

---

This deployment guide shows you how to set up an Amazon Elastic Kubernetes

> Engine "EKS" cluster on which Coder can deploy.

Reviewer: ```suggestion
Service (EKS) cluster on which Coder can deploy.
```

  to fast-track this process
- [eksctl command-line utility](https://docs.aws.amazon.com/eks/latest/userguide/eksctl.html)

> See [Preliminary Steps](#preliminary-steps) and [Node Considerations](#node-considerations) first to familiarize yourself with steps and items before creating a cluster.

Reviewer: ```suggestion
Before you create a cluster, see first [Preliminary Steps](#preliminary-steps) and [Node Considerations](#node-considerations).
```

## Step 1: Create an EKS cluster

> While flags can be passed to `eksctl create cluster`, the following example uses
> a configuration yaml file to define an EKS cluster.

Reviewer: ```suggestion
While flags can be passed to `eksctl create cluster`, the following example uses
an [`eksctl` configuration file](https://eksctl.io/usage/schema/) to define an EKS cluster.
```

> > The cluster name,

Reviewer: ```suggestion
> The cluster name
```

[region](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html#concepts-regions),
and SSH key path will be specific to your installation so potentially change
them in the yaml file

```yaml
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: trial-cluster
  region: us-east-1

managedNodeGroups:
  - name: managed-ng-1
    instanceType: t2.medium
    amiFamily: Ubuntu2004
    desiredCapacity: 1
    minSize: 1
    maxSize: 2
    volumeSize: 50
    ssh:
      allow: true

>       publicKeyPath: ~/.ssh/id_rsa.pub

Reviewer: ```suggestion
      publicKeyPath: path/to/you/rs
```

```

```console
eksctl create cluster -f cluster.yaml
```

> If your developers require Docker commands like `docker build`, `docker run`, and `docker-compose` as
> part of their development flow, then [container-based virtual machines

> > (CVMs)](../../workspaces/cvms.md) are required. `instantType` of `Ubuntu2004`
> > is preferred since `AmazonLinux2` does not support caching and the shiftfs
> > kernel module in CVM settings.

Reviewer: ```suggestion
> (CVMs)](../../workspaces/cvms.md) are required.  In this case, we recommend using the
> `Ubuntu2004` AMI family, as the `AmazonLinux2` AMI family does not meet the requirements
> for [cached CVMs](../../workspace-management/cvms/management#caching).
```

1. Now that you have the `.pem` file, extract the public key portion of the
   keypair so that you can use it with the eksctl CLI in later steps:

   ```sh
   ssh-keygen -y -f <PATH/TO/KEY>.pem >> <PATH/TO/KEY/KEY>.pub
   ```

   **Note**: if you run into a bad permissions error, run `sudo` before the
   command above.

When done, you should have a .pem and .pub file for the same keypair you
downloaded from AWS.

## Node Considerations

The node type and size that you select impact how you use Coder. When choosing,
be sure to account for the number of developers you expect to use Coder, as well
as the resources they need to run their workspaces. See our guide on on
[compute resources](../../guides/admin/resources.md) for additional information.

If you expect to provision GPUs to your Coder workspaces, you **must** use an
EC2 instance from AWS'
[accelerated computing instance family](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/accelerated-computing-instances.html).

> GPUs are not supported in workspaces deployed as
> [container-based virtual machines (CVMs)](../../workspaces/cvms.md) unless
> you're running Coder in a bare-metal Kubernetes environment.

> 

Reviewer: Suggestion: move this to the top of the document before step 1 for readability.
